source/car/electronics_embedded_systems/embedded_modules/embedded_modules.rst

```diff
@@ -0,0 +1,598 @@
+Embedded Modules
+================
+
+Seven embedded modules provide autonomous capabilities to the self-driving vehicle.
+
+.. figure:: /images/electronics_embedded/sdv_systems.png
+   :align: center
+   :alt: sdv_systems
+   :figclass: align-center
+   :width: 800px
+   
+   SDV Modules, Sensors and Computing Devices
+
+|
+
+Common Module Subsystems
+========================
+SDV modules are similar (with slight variations) in certain subsystems, such as in IO, and power supply.
+
+-----
+STM32
+-----
+
+The `STM32L431RCT6 <https://www.lcsc.com/product-detail/Microcontroller-Units-MCUs-MPUs-SOCs_STMicroelectronics-STM32L431RCT6_C92468.html>` is chosen as the main microcontroller unit for all modules
+to follow an standard started by the development of the USV and UUV PCBs, where a similiar version was used, which in turn eases the electronics and embedded development.
+
+.. figure:: /images/electronics_embedded/stm32_base.png
+   :align: center
+   :alt: stm32 schematic
+   :figclass: align-center
+   :width: 600px
+   
+   Base STM32 configuration
+
+|
+
+High-Speed External Clock
+-------------------------
+.. figure:: /images/electronics_embedded/hsec.png
+   :align: center
+   :alt: hsec
+   :figclass: align-center
+   :width: 200px
+   
+   Oscillator circuit
+
+`This <https://www.st.com/resource/en/application_note/an2867-oscillator-design-guide-for-stm8afals-stm32-mcus-and-mpus-stmicroelectronics.pdf>` design guide was used to choose the crystal circuit components.
+
+The high-speed external clock is supplied with a 16MHz as the high-speed external crystal oscillator.
+
+The `crystal <https://www.lcsc.com/product-detail/Crystals_Yangxing-Tech-X322516MLB4SI_C13738.html>` has a load capacitance (CL) of 9pF, and assuming a stray PCB capacitance (Cs) of 3pF, 12pF capacitors (CL1 and CL2) were chosen following the next equation.
+
+.. figure:: /images/electronics_embedded/crystal_eq.png
+   :align: center
+   :alt: hsec
+   :figclass: align-center
+   :width: 200px
+   
+   Crystal equation
+
+|
+
+
+STM32 Programming, Supply and Communications
+--------------------------------------------
+An ST-Link is used to program the STM32 via a 10 pin header.
+
+A 5-pin M12 connector is used to supply +12V, and to connect the module to the CAN Bus. The 5th pin is left unused.
+
+.. figure:: /images/electronics_embedded/prog.png
+   :align: center
+   :alt: hsec
+   :figclass: align-center
+   :width: 400px
+   
+   Programming, Supply and Comms
+
+|
+
+Module ID
+---------
+Each module can be mannually assigned an ID, which can be used fpr the CAN messages identification. A maximum of 16 IDs can be assigned.
```

Comment:
*for CAN messages ...

Reply:
eso en otro pull request, todavia andamos checando los ids del encoder etc etc, para hacer eso una sola vez